src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/RecommendCaseInsensitiveStringComparison.Analyzer.cs

```diff
@@ -0,0 +1,188 @@
+                return;
+            }
+
+            // Retrieve the diagnosable string overload methods: Contains, IndexOf, StartsWith, CompareTo
+
+            ParameterInfo[] stringParameter = new[]
+            {
+                ParameterInfo.GetParameterInfo(stringType)
+            };
+
+            IMethodSymbol? containsStringMethod = stringType.GetMembers(StringContainsMethodName).OfType<IMethodSymbol>().GetFirstOrDefaultMemberWithParameterInfos(stringParameter);
+            if (containsStringMethod == null)
+            {
+                return;
+            }
+
+            // TODO: There are more overloads that take StringComparison, diagnose only the simple one for now
+            IMethodSymbol? indexOfStringMethod = stringType.GetMembers(StringIndexOfMethodName).OfType<IMethodSymbol>().GetFirstOrDefaultMemberWithParameterInfos(stringParameter);
+            if (indexOfStringMethod == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? startsWithStringMethod = stringType.GetMembers(StringStartsWithMethodName).OfType<IMethodSymbol>().GetFirstOrDefaultMemberWithParameterInfos(stringParameter);
+            if (startsWithStringMethod == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? compareToStringMethod = stringType.GetMembers(StringCompareToMethodName).OfType<IMethodSymbol>().GetFirstOrDefaultMemberWithParameterInfos(stringParameter);
+            if (compareToStringMethod == null)
+            {
+                return;
+            }
+
+            // Retrieve the StringComparer properties that need to be flagged: CurrentCultureIgnoreCase, InvariantCultureIgnoreCase
+
+            IEnumerable<IPropertySymbol> ccicPropertyGroup = stringComparerType.GetMembers(StringComparisonCurrentCultureIgnoreCaseName).OfType<IPropertySymbol>();
+            if (!ccicPropertyGroup.Any())
+            {
+                return;
+            }
+
+            IEnumerable<IPropertySymbol> icicPropertyGroup = stringComparerType.GetMembers(StringComparisonInvariantCultureIgnoreCaseName).OfType<IPropertySymbol>();
+            if (!icicPropertyGroup.Any())
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(context =>
+            {
+                IInvocationOperation caseChangingInvocation = (IInvocationOperation)context.Operation;
+                IMethodSymbol caseChangingMethod = caseChangingInvocation.TargetMethod;
+
+                if (!caseChangingMethod.Equals(toLowerParameterlessMethod) &&
+                    !caseChangingMethod.Equals(toLowerInvariantParameterlessMethod) &&
+                    !caseChangingMethod.Equals(toUpperParameterlessMethod) &&
+                    !caseChangingMethod.Equals(toUpperInvariantParameterlessMethod))
+                {
+                    return;
+                }
+
+                if (caseChangingInvocation.Parent is not IInvocationOperation diagnosableInvocation)
```

Comment:
Quite uncommon case, but should we walk up parenthesized operations first before checking the diagnosable invocation?

Reply:
Can you give me an example?

Reply:
@carlossanlop `int r = (a.ToLower()).CompareTo(b);`

Reply:
Ah good point! Let me test this and determine how complex it would be to diagnose a case like this.

Reply:
Interesting, the code works as it is without any additional changes. I'll push the test so you can see what I mean.

Reply:
Which means I was inadvertently removing parenthesis. I am trying to think of a case where the parenthesis _need_ to be preserved.

Reply:
Ah, it only works for C#, but it doesn't work for VB.

Reply:
Fixed!